### ISSUE TITLE
Dropdown: Preselected value not shown when using reactive forms

### ISSUE THREAD

**Issue by Timmeeeey** [primefaces/primeng#14241]
### Describe the bug

When using reactive forms with an initial value this value is not shown in the dropdown.
It shows only after clicking on the dropdown.

![2023-11-29_10h02_26](https://github.com/primefaces/primeng/assets/46870283/5db224a7-fb0b-4bde-9d2f-db5d244cc2ca)

### Environment

Windows, Edge

### Reproducer

https://stackblitz.com/edit/z2vojt

### Angular version

17.0.0

### PrimeNG version

17.0.0-beta.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

20.9.0

**Comment by jerkovicl** [comment]
@Timmeeeey same issue here
if your value is set during creating of form like this: new FormControl(value here), dropdown will not show selected value initially, it will only show it after dropdown click

**Comment by jerkovicl** [comment]
@Timmeeeey  another thing, if you remove `?` from `selectedItem` template value will be shown in dropdown but we will get error in console about item being `null` in `selectedItem` template
reproducer [here:](https://stackblitz.com/edit/z2vojt-rpxuv9?file=src%2Fapp%2Fdemo%2Fdropdown-template-demo.html)

**Comment by chaeller** [comment]
I have the problem if I use filters:

17.0.0-beta.1 = not OK
  https://stackblitz.com/edit/uzpr21-k9k6xy
  or
  https://stackblitz.com/edit/uzpr21-dhskbu

16.9.1 = OK
  https://stackblitz.com/edit/cnmrtn-bw3sgn

**Comment by jerkovicl** [comment]
@cetincakiroglu any updates on this? did you check reproducers?

**Comment by jerkovicl** [comment]
@cetincakiroglu  tested with latest version `v17.0.0`, this is still not working, [reproducer](https://stackblitz.com/edit/z2vojt-rpxuv9?file=src%2Fapp%2Fdemo%2Fdropdown-template-demo.html)

**Comment by aturoczy** [comment]
I have the same issue. Last 2 days I tried to debug it, and finally found this thread :) The only "workaround" that I found to bind the value into the ngModel as well. But it is definitely a breaking change that the Angular 17 brought.

**Comment by jerkovicl** [comment]
@cetincakiroglu this should be reopened, since issue is still there even with 17.1.0 fix 
[Reproducer](https://stackblitz.com/edit/z2vojt-rpxuv9?file=package.json)

**Comment by aturoczy** [comment]
I'm going to try it soon with the new version. But yes, in the release notes this fix is there.

**Comment by cetincakiroglu** [comment]
> @cetincakiroglu this should be reopened, since issue is still there even with 17.1.0 fix [Reproducer](https://stackblitz.com/edit/z2vojt-rpxuv9?file=package.json)
![Screenshot 2023-12-13 at 19 39 38](https://github.com/primefaces/primeng/assets/69278826/aa16905c-d852-40d9-8a84-ea8bd1874817)

are you sure? your example is working actually...

**Comment by jerkovicl** [comment]
> > @cetincakiroglu this should be reopened, since issue is still there even with 17.1.0 fix [Reproducer](https://stackblitz.com/edit/z2vojt-rpxuv9?file=package.json)
> 
> ![Screenshot 2023-12-13 at 19 39 38](https://github.com/primefaces/primeng/assets/69278826/aa16905c-d852-40d9-8a84-ea8bd1874817)
> 
> 
> 
> Are you sure? Your example is working actually, I didn't change anything just opened the example you've shared above.

Check the error in console

**Comment by cetincakiroglu** [comment]
> > > @cetincakiroglu this should be reopened, since issue is still there even with 17.1.0 fix [Reproducer](https://stackblitz.com/edit/z2vojt-rpxuv9?file=package.json)
> > 
> > 
> > ![Screenshot 2023-12-13 at 19 39 38](https://private-user-images.githubusercontent.com/69278826/290263364-aa16905c-d852-40d9-8a84-ea8bd1874817.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0ODkzMzQsIm5iZiI6MTcwMjQ4OTAzNCwicGF0aCI6Ii82OTI3ODgyNi8yOTAyNjMzNjQtYWExNjkwNWMtZDg1Mi00MGQ5LThhODQtZWE4YmQxODc0ODE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEzVDE3MzcxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEzMGUzNjZkZGY3OGYyMzc3OGI0MzYyZmUzNmM2YzcyY2U3YTc0YTY3MDRkOGNkMWU1NzU0M2ZjM2I0MjE1YzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WFxOQxwJlUWx9bIHLJLgATMI0IpeetH32ThOksYBNSI)
> > Are you sure? Your example is working actually, I didn't change anything just opened the example you've shared above.
> 
> Check the error in console

It's another issue, I'll create a new issue for it, thanks a lot for reporting

**Comment by jerkovicl** [comment]
@cetincakiroglu ok thats fine

**Comment by rburgst** [comment]
I still have occasional instances of the pre-selected item not showing up, however, its not trivial to reproduce.

**Comment by rburgst** [comment]
the problem seems to happen when initially the list of options if `undefined`. Will try and create a repro

**Comment by yassine-fy94** [comment]
The filterBy option is not working...

**Comment by xeexthemoth** [comment]
I've found a crazy work around... if you add virtual scroll to the component then it works!
e.g.
<p-dropdown 
..
..
..
[virtualScroll]="true"
[virtualScrollItemSize]="10"
>

**Comment by xeexthemoth** [comment]
found a better hack!
 last thing in your init

            setTimeout(() => {
                const originalValue = this.myForm.controls['myDropDownName'].value;
                this.myForm.controls['myDropDownName'].setValue('');
                this.myForm.controls['myDropDownName'].setValue(originalValue);
            });
            
seems to need to set value to an empty string first otherwise it doesn't work... so very strange... it is like the dropdown isn't updating its on screen paint once everything has completed... maybe a zone file bug in angular or a weird bug in primeng dropdown...

**Comment by GrantLiang** [comment]
![image](https://github.com/primefaces/primeng/assets/8802427/592b7674-0824-4ebe-a8de-23c7ed65c9d7)

https://stackblitz.com/edit/xdzkrc

In the above example, using Float Label
but No.3 case push item into options then the selected option will disappear

primeng : v17.14.1
